### PR TITLE
Allow for request models to define their item limits (defaults to nil)

### LINKED ISSUE #122: Add configurable item selector limit
Everybody can have unlimited items
SPEC-COLL: Limit 5
Scan request: Limit 1

### PR DESCRIPTION
**Pull request by jkeck**
Closes #122 

- [x] Scan: Limit 1
- [x] SPEC-COLL Mediated Page: Limit 5
- [x] Everything else: Unlimited

### CODE REVIEW

File: app/assets/javascripts/limit_selected_items.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
   $.fn.limitSelectedItems = function() {
     this.each(function(){
       var itemSelector = $(this);
-      var limit = 5;
+      var limit = itemSelector.data('limit-selected-items');
```
Comment (jchristo4) [inline app/assets/javascripts/limit_selected_items.js]: @jkeck A minor data parsing step here for safety -
```javascript
var limit = parseInt(itemSelector.data('limit-selected-items'), 10);
```

Comment (jkeck) [inline app/assets/javascripts/limit_selected_items.js]: :+1: thanks @jchristo4.  I have made that change.

**Comment by jgreben** [comment]
For Scans (and perhaps Mediated pages) when you have a item checked and you refresh the page, you are able to select another `n` items according to the defined limits (a turbolinks issue?)

**Comment by jkeck** [comment]
@jgreben please create a **New** issue to report this.

**Pull request merged**